hokusai/commands/push.py

```diff
@@ -1,42 +1,60 @@
-import os
-
 from hokusai.lib.command import command
 from hokusai.lib.config import config
 from hokusai.services.ecr import ECR
-from hokusai.lib.common import print_green, shout
+from hokusai.lib.common import print_green, print_yellow, shout
 from hokusai.lib.exceptions import HokusaiError
 from hokusai.services.docker import Docker
 
-@command()
-def push(tag, local_tag, build, filename, force, overwrite, skip_latest=False):
-  if force is None and shout('git status --porcelain'):
-    raise HokusaiError("Working directory is not clean.  Aborting.")
-
-  if force is None and shout('git status --porcelain --ignored'):
-    raise HokusaiError("Working directory contains ignored files and/or directories.  Aborting.")
+def build_and_push(remote_tag, local_tag, filename, ecr, skip_latest=False):
+  """Build Docker image and push it to ECR"""
+  Docker().build(filename)
+  push_only(remote_tag, local_tag, ecr, skip_latest)
 
+def ecr_check():
+  """Check that ECR has a repo for the project"""
   ecr = ECR()
   if not ecr.project_repo_exists():
-    raise HokusaiError("ECR repo %s does not exist... did you run `hokusai setup` for this project?" % config.project_name)
-
-  shout(ecr.get_login(), mask=(r'^(docker login -u) .+ (-p) .+ (.+)$', r'\1 ****** \2 ***** \3'))
-  if tag is None:
-    tag = shout('git rev-parse HEAD').strip()
-
-  if overwrite is None and ecr.tag_exists(tag):
-    raise HokusaiError("Tag %s already exists in registry.  Aborting." % tag)
+    raise HokusaiError("ECR repo %s does not exist. Has 'hokusai setup' been run for this project?" % config.project_name)
+  return ecr
+
+def git_status_check():
+  """Check working directory's Git status"""
+  git_status_files = shout('git status --porcelain --ignored')
+  if git_status_files:
+    print_yellow("The following files/directories are either changed, untracked, or Git ignored.")
+    print_yellow("If they are not excluded by .dockerignore, they may get copied into the resulting Docker image.")
+    print_yellow(git_status_files)
+    raise HokusaiError("Working directory unclean. Aborting. Use '--force' to force.")
 
+@command()
+def push_image(remote_tag, local_tag, build, filename, force, overwrite, skip_latest=False):
```

Comment:
`push.py` file is interpreted as a module by Python with module name `push`. When this function is named `push` and [imported in __init__py](https://github.com/artsy/hokusai/blob/d5f2cd3f5125c90041c5ea3a897d8e9245994200/hokusai/commands/__init__.py#L13), the function masks the module, so that the module cannot be imported elsewhere. Well, for the tests I have to [import the module](https://github.com/artsy/hokusai/pull/346/files#diff-796b61f71f490c9a3dd9fd895cc4133a85136b4c620a6a1102adb60da0fbb297R3), so renaming the function to `push_image`.